app/src/main/java/ch/hikemate/app/model/authentication/AuthViewModel.kt

```diff
@@ -25,12 +28,19 @@ class AuthViewModel(private val repository: FirebaseAuthRepository) {
    * @param coroutineScope The CoroutineScope in which this operation will be executed.
    * @param context The context is used to start the Google Sign-In process.
    */
-  fun signInWithGoogle(coroutineScope: CoroutineScope, context: Context) {
+  fun signInWithGoogle(
+      coroutineScope: CoroutineScope,
+      context: Context,
+      startAddAccountIntentLauncher: ManagedActivityResultLauncher<Intent, ActivityResult>?,
```

Comment:
Question: Why is the `startAddAccountIntentLauncher` nullable?

Reply:
I wrote a possible implementation of the launcher while testing my functionality, and it looked something like this:
```kotlin val addAccountLauncher =

          rememberLauncherForActivityResult(
            contract = ActivityResultContracts.StartActivityForResult()) {
              result: ActivityResult ->
            authViewModel.signInWithGoogle(coroutineScope, context, null)
            Toast.makeText(context, "success", Toast.LENGTH_SHORT).show()
            Log.d("MainActivity", "addAccountLauncher result: $result")
          }
```

This `rememberLauncherForActivityResult` is only callable from a composable, so in the UI. This is why I decided to pass it all the way down... If you know a nicer way to solve this I'd be happy to hear, I also don't love this solution.

Reply:
Yep, okay, I see, but I think you answered the wrong comment!

Reply:
Ah, my bad, the launcher is nullable since after connecting the account together was successful, we make the call to the authViewModel that I also show in my answer. 
```kotlin 
authViewModel.signInWithGoogle(coroutineScope, context, null)`
If we the launcher was not nullable, we would have to pass the addAccountLauncher itself as parameter here, which does not work afaik. It being null is not an issue however, since the NoCredentialError should be resolved after the google sign in is completed, so the addAccountLauncher, which is only ever used when a NoCredentialError is catched, should not need to be called.

Reply:
Mmmmh, I see. Definitely not ideal, but I don't see any way to improve it without a whole refactor 😅. So leave it like it